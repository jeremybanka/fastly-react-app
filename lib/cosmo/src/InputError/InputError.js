/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Text } from "../";

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type InputErrorProps = {
  htmlFor: string,
  children: TextType,
};
const InputError = (props: InputErrorProps) => {
  return (
    <Text as="label" htmlFor={props.htmlFor} color="reds.5" fontSize="xxs">
      {props.children}
    </Text>
  );
};

export default InputError;
