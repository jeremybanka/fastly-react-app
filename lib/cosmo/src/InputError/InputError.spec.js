/* @flow */

import React from "react";
import { render } from "enzyme";
import InputError from "./InputError";

describe("InputError test suite", () => {
  const component = <InputError>Error</InputError>;
  it("renders the stored snapshot", () => {
    expect(render(component)).toMatchSnapshot();
  });
});
