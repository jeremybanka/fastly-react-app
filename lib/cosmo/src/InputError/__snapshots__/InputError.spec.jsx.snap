// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError test suite renders the stored snapshot 1`] = `
<label
  class="Text-sc-6c1xd4-0 frfOcY"
  color="reds.5"
  font-size="xxs"
>
  Error
</label>
`;
